modules/platform/nuxeo-collections/src/test/java/org/nuxeo/ecm/collections/core/test/CollectionCreateFolderVersionTest.java

```diff
@@ -0,0 +1,104 @@
+ *
+ * Contributors:
+ *     Charles Boidot
+ */
+
+package org.nuxeo.ecm.collections.core.test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.collections.api.CollectionConstants;
+import org.nuxeo.ecm.collections.api.CollectionManager;
+import org.nuxeo.ecm.collections.core.adapter.Collection;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelIterator;
+import org.nuxeo.ecm.core.api.VersioningOption;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+@RunWith(FeaturesRunner.class)
+@Features({ CoreFeature.class, CollectionFeature.class})
+public class CollectionCreateFolderVersionTest {
+
+    @Inject
+    CollectionManager collectionManager;
+
+    @Inject
+    CoreSession session;
+
+
+    private static final String TEST_FOLDER = "testFolder";
+
+    private static final String TEST_COLLECTION_IN_FOLDER = "testCollectionInFolder";
+
+    protected static final String TEST_FILE_NAME = "testFile";
+
+    protected static final int NB_TEST_FILES = 11;
+
+    @Test
+    public void testCreateFolderVersionContainingACollection() throws Exception {
+        List<DocumentModel> files = new ArrayList<>();
+        DocumentModel testWorkspace = session.createDocumentModel("/default-domain/workspaces", "testWorkspace",
+                "Workspace");
+        testWorkspace = session.createDocument(testWorkspace);
+        DocumentModel folder = session.createDocumentModel(testWorkspace.getPathAsString(), TEST_FOLDER,
+                "Folder");
+        folder = session.createDocument(folder);
+
+        for (int i = 1; i <= NB_TEST_FILES; i++) {
+            DocumentModel testFile = session.createDocumentModel(folder.getPath().toString(), TEST_FILE_NAME + i,
+                    "File");
+            testFile = session.createDocument(testFile);
+            files.add(testFile);
+        }
+
+        DocumentModel collectionInFolder = session.createDocumentModel(folder.getPathAsString(),
+                TEST_COLLECTION_IN_FOLDER, CollectionConstants.COLLECTION_TYPE);
+        collectionInFolder = session.createDocument(collectionInFolder);
+        session.save();
+        collectionManager.addToCollection(collectionInFolder, files, session);
+        Collection collectionAdapter = collectionInFolder.getAdapter(Collection.class);
+
+        //Check all children files are in the collection
+        assertEquals(NB_TEST_FILES,collectionAdapter.getCollectedDocumentIds().size());
+
+        // Create a new version of the folder
+        DocumentModel newVersionFolder = session.getDocument(session.checkIn(folder.getRef(), VersioningOption.MAJOR, ""));
```

Comment:
`newVersionFolder `never used, calling `checkIn` without reading the result is enough for this test